gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -2028,3 +2034,113 @@ def test_{{transport_name}}_initialize_client():
         {{ await_prefix }}client.{{ method_name }}(request)
 
 {% endmacro %}
+
+{# rest_method_call_success_test generates tests for rest methods
+ # when they make a successful request.
+ # NOTE: Currently, this macro does not support the following method
+ # types: [method.server_streaming, method.lro, method.extended_lro, method.paged_result_field].
+ # As support is added for the above methods, the relevant guard can be removed from within the macro
+ # TODO(https://github.com/googleapis/gapic-generator-python/issues/2142): Clean up `rest_required_tests` once support for all the methods metioned above is added here.
+#}
+{% macro rest_method_call_success_test(service, method, method_name, transport_name, async_prefix, await_prefix, async_decorator, is_async) %}
+
+{# NOTE: set method_output to method.extended_lro.operation_type for the following method types:
+ # (method.extended_lro and not full_extended_lro)
+#}
+{% set method_output = method.output %}
+{# TODO(https://github.com/googleapis/gapic-generator-python/issues/2143): Update the guard below as support we add for each method. Remove it once we have
+ # all the methods supported in async rest transport that are supported in sync rest transport.
+ #}
+{% if not (method.server_streaming or method.lro or method.extended_lro or method.paged_result_field)%}
+@pytest.mark.parametrize("request_type", [
+  {{ method.input.ident }},
+  dict,
+])
+{{async_decorator}}
+{{async_prefix}}def test_{{method_name}}_{{transport_name}}_call_success(request_type):
```

Comment:
Could the `method_call_test` from https://github.com/googleapis/gapic-generator-python/pull/2126/files be used here to reduce code duplication?

Reply:
I'm not sure if it can exactly be used as is but as I'm working through refactoring the test cases, I see that we can reduce code duplication by pulling out the logic to generate mock response according to the type of api method being tested into a separate macro.

If there's obvious code duplication that we're introducing or if sth could be easily resolved then we shall address that. Otherwise, I think it's alright to do this as a follow up. WDYT?